libraries/abstractions/common_io/include/iot_spi.h

```diff
@@ -155,183 +163,259 @@ void iot_spi_set_callback( IotSPIHandle_t const pxSPIPeripheral,
+ *
+ * This function attempts to write certain number of bytes from a pre-allocated buffer to a slave device, in synchronous way.
+ * This function does not return on paritial write, unless there is an error.
+ * And the number of bytes that have been actually written can be obtained by calling iot_spi_ioctl.
  *
- * The application should call this function to start synchronous write.
- * The application should also give the address of the buffer to send the data
- * from; with number of bytes to write.
+ *
+ * @warning None of other read or write functions shall be called during this function.
  *
  * @param[in] pxSPIPeripheral The SPI peripheral handle returned in open() call.
  * @param[in] pvBuffer The buffer with data to transmit.
  * @param[in] xBytes The number of bytes to write.
  *
- * @return SUCCESS = IOT_SPI_SUCCESS,
- *          FAILED = IOT_SPI_INVALID_VALUE,
- *          SPI_BUSY = IOT_SPI_BUS_BUSY,
- *          Master_UNINITIALIZED = IOT_SPI_MASTER_NOT_INITIALIZED
- *          Slave not selected = IOT_SPI_SLAVE_NOT_SELECTED
+ * @return
+ * - IOT_SPI_SUCCESS, on success (all the requested bytes have been read)
+ * - IOT_SPI_INVALID_VALUE, if
+ *     - pxSPIPeripheral is NULL
+ *     - pxSPIPeripheral is not opened yet
+ *     - pucBuffer is NULL
+ *     - xBytes is 0
+ * - IOT_SPI_WRITE_ERROR, if there is some unknown driver error.
+ * - IOT_SPI_BUSY, if the bus is busy which means there is an ongoing transaction.
  */
 int32_t iot_spi_write_sync( IotSPIHandle_t const pxSPIPeripheral,
                             uint8_t * const pvBuffer,
                             size_t xBytes );
 
 /**
- * @brief The SPI master starts transmission of data to the slave
- * asynchronously.
+ * @brief The SPI master starts transmission of data to the slave asynchronously.
+ *
+ * This function attempts to read certain number of bytes from a pre-allocated buffer, in asynchronous way.
+ * It returns immediately when the operation is started and the status can be check by calling iot_spi_ioctl.
+ * Once the operation completes, successful or not, the user callback will be invoked.
  *
- * The application should call this function to start asynchronous write.
- * The application should also give the address of the buffer to send the data
- * from; with number of bytes to write.
+ * Once the operation completes successfully, the user callback will be invoked.
+ * If the operation encounters an error, the user callback will be invoked.
+ * The callback is not invoked on paritial write, unless there is an error.
+ * And the number of bytes that have been actually read can be obtained by calling iot_spi_ioctl.
+ *
+ * @note In order to get notification when the asynchronous call is completed, iot_spi_set_callback must be called prior to this.
+ *
+ * @warning pucBuffer must be valid before callback is invoked.
+ * @warning None of other read or write functions shall be called during this function or before user callback.
```

Comment:
This is general comment that applies to all he write and read calls. Since HAL does not support concurrent calls, if a read or write is in progress - the next calls to read/write shall return BUS_BUSY.  If we want to add this, then this needs to be added to all the write/read calls. 

Also - please consider rewording to "when a write_async is in progress, no other read or write calls can be made. The sub-sequent calls to any read or write APIs will return BUS_BUSY error"

Reply:
I am OK to remove this warning because we already say IOT_SPI_BUSY is returned if the bus is busy which means there is an ongoing operation

Reply:
ok